cluster-autoscaler/core/podlistprocessor/filter_out_schedulable.go

```diff
@@ -29,16 +29,19 @@ import (
 	"k8s.io/autoscaler/cluster-autoscaler/simulator/scheduling"
 	corev1helpers "k8s.io/component-helpers/scheduling/corev1"
 	klog "k8s.io/klog/v2"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
 )
 
 type filterOutSchedulablePodListProcessor struct {
 	schedulingSimulator *scheduling.HintingSimulator
+	nodesToConsider     func(*schedulerframework.NodeInfo) bool
 }
 
 // NewFilterOutSchedulablePodListProcessor creates a PodListProcessor filtering out schedulable pods
-func NewFilterOutSchedulablePodListProcessor(predicateChecker predicatechecker.PredicateChecker) *filterOutSchedulablePodListProcessor {
+func NewFilterOutSchedulablePodListProcessor(predicateChecker predicatechecker.PredicateChecker, nodesToConsider func(*schedulerframework.NodeInfo) bool) *filterOutSchedulablePodListProcessor {
```

Comment:
nit: `nodesToConsider` sounds like it would be a slice, not a function.

Maybe sth like `nodeFilter`? Since it essentially filters out nodes we don't want to use for this simulation, regardless of the properties of the pod being considered.

Reply:
Done